### PR TITLE
Basic implementation of Java to JS proxy for QuickJs.

### PR DESCRIPTION
**Pull request by szurbrigg**

### CODE REVIEW

**Comment by szurbrigg** [comment]
No argument support, and the internal proxy structures are leaking, but the basics are there.

File: quickjs/src/androidTest/java/com/squareup/quickjs/QuickJsGetTest.java [hunk 1]
```diff
@@ -0,0 +1,593 @@
+/*
```
Comment (szurbrigg) [inline quickjs/src/androidTest/java/com/squareup/quickjs/QuickJsGetTest.java]: Pretty much a copy of DuktapeGetTest with some renames.

File: quickjs/src/main/java/com/squareup/quickjs/QuickJs.java [hunk 2]
```diff
@@ -66,6 +70,57 @@ public synchronized Object evaluate(@NonNull String script) {
     return evaluate(context, script, "?");
   }
 
+  /**
```
Comment (szurbrigg) [inline quickjs/src/main/java/com/squareup/quickjs/QuickJs.java]: Copied from Duktape.java

File: quickjs/src/androidTest/java/com/squareup/quickjs/QuickJsGetTest.java [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.fail;
+
+public final class QuickJsGetTest {
+    private QuickJs quickJs;
```
Comment (JakeWharton) [inline quickjs/src/androidTest/java/com/squareup/quickjs/QuickJsGetTest.java]: two-space indent!

if you need it: https://github.com/square/java-code-styles

File: quickjs/src/main/jni/quickjs-jni.c [hunk 4]
```diff
@@ -74,87 +117,248 @@ Java_com_squareup_quickjs_QuickJs_destroyContext(JNIEnv *env, jclass type, jlong
...
+                                                                                    jlong context_,
+                                                                                    jstring sourceCode_,
+                                                                                    jstring fileName_) {
+  Context* context = (Context*) context_;
+  JSContext* jsContext = context->jsContext;
+
+  const char* sourceCode = (*env)->GetStringUTFChars(env, sourceCode_, 0);
+  const char* fileName = (*env)->GetStringUTFChars(env, fileName_, 0);
+
+  JSValue evalValue = JS_Eval(jsContext, sourceCode, strlen(sourceCode), fileName, 0);
```
Comment (swankjesse) [inline quickjs/src/main/jni/quickjs-jni.c]: the asymmetry here will always weird me out

**Comment by swankjesse** [comment]
Supercool.

**Pull request merged**